### ISSUE TITLE
Caffe to IR grabs concat axis from wrong parameter

### ISSUE THREAD

**Issue by oveddan** [Microsoft/MMdnn#155]
## The issue

I'm trying to convert the pre-trained [Eye Tracking for Everyone caffe model](https://github.com/CSAILVision/GazeCapture/tree/master/models) to Tensorflow.

As an immediate step, this needs to be converted first to IR. In this step, the `concat` layer's axis is not grabbed correctly from the `.prototxt` file in this conversion process.

If you look at the code for the [caffe_emitter](https://github.com/Microsoft/MMdnn/blob/master/mmdnn/conversion/caffe/caffe_emitter.py#L307) you see that it is grabbing concat axis using:

```python
axis = axis_array.index(IR_node.get_attr('axis'))
```

Really the axis is stored in `concat_param`, not in `param`.

See the [caffe documentation for concat](http://caffe.berkeleyvision.org/tutorial/layers/concat.html) and the [pre-trained model to convert](https://github.com/CSAILVision/GazeCapture/blob/master/models/itracker_deploy.prototxt#L641):

```json
layer {
  name: "concat2"
  bottom: "fc1"
  bottom: "fg_fc2"
  bottom: "fc2_f"
  top: "concat2"
  type: "Concat"
  concat_param {
    axis: 1
  }
}
```

## Reproducing the error

If you run the code:
```bash
git clone https://github.com/CSAILVision/GazeCapture.git

mkdir -p ./models/

mmtoir -f caffe -n "./GazeCapture/models/itracker_deploy.prototxt" -w "./GazeCapture/models/snapshots/itracker25x_iter_92000.caffemodel" \
 -o ./models/caffe_itracker_IR
```

Then view the contents of `./models/caffe_itracker_IR.json` you see that the axis for `concat1` and `concat2` are 3, instead of 1, which is what they should be:

```shell
cat ./models/caffe_itracker_IR.json
```

Produces  `concat1`:

```json
...
{
            "attr": {
                "_output_shapes": {
                    "list": {
                        "shape": [
                            {
                                "dim": [
                                    {
                                        "size": "-1"
                                    },
                                    {
                                        "size": "13"
                                    },
                                    {
                                        "size": "13"
                                    },
                                    {
                                        "size": "128"
                                    }
                                ]
                            }
                        ]
                    }
                },
                "axis": {
                    "i": "3"
                }
            },
            "input": [
                "relu4_l",
                "relu4_r"
            ],
            "name": "concat1",
            "op": "Concat"
        }
...
...
{
            "attr": {
                "_output_shapes": {
                    "list": {
                        "shape": [
                            {
                                "dim": [
                                    {
                                        "size": "-1"
                                    },
                                    {
                                        "size": "1"
                                    },
                                    {
                                        "size": "1"
                                    },
                                    {
                                        "size": "320"
                                    }
                                ]
                            }
                        ]
                    }
                },
                "axis": {
                    "i": "3"
                }
            },
            "input": [
                "relufc1",
                "relufgfc2",
                "relufc2_f"
            ],
            "name": "concat2",
            "op": "Concat"
        }
```

The axis for both are 3 when they should be 1.

**Comment by kitstar** [comment]
Because the IR is channel first format. Axis 1 in caffe and axis 3 in IR means the channel axis.

**Comment by kitstar** [comment]
Oh, got it. It's hard to make the decision if we need to transfer the axis, since there is no enough output spatial information in Caffe...

And thanks for using MMdnn!
